packages/transport/src/modules/tokens/local/Token.ts

```diff
@@ -50,7 +50,8 @@ export class Token extends CoreSynchronizable implements IToken {
     }
 
     public toTokenReference(): TokenReference {
-        return TokenReference.from({ id: this.id, key: this.secretKey, forIdentity: this.cache?.forIdentity });
+        // TODO: add the token reference forIdentity once available
```

Comment:
no todo comments in code we want to merge please.

Reply:
I converted the PR to a draft as I would highly recommend adding this feature back so we can catch wrong personalized tokens before sending them to the backend and getting that error.

Backend error will be and 404 Not found. And we could send "Not intended for you"

Reply:
We should not handle this task like this.

The extension of the truncated reference should only be an optimization to intercept early when you don't have access to the token. As the trunacted reference is not the only way to load tokens we should find other ways to make this secure - for example going back to the backbone team and asking for more interfaces / routes to make it usable.

Reply:
My original idea was that the backbone would send a different error in this case, I think a 403 would be appropriate

Reply:
yes, we should request that.

Reply:
Also for example a route like /canAccess could be good.

Reply:
@tnotheis also had the idea to not differentiate between no token and not correctly personalized token for security purposes. Also why do we offer multiple ways to load tokens?

Reply:
This would also be in line with what we currently have for other objects. For example, if you have the ID of an existing message of which you are not a participant (-> neither sender nor recipient), you will receive a 404 without a specific reason.

Reply:
yeah, but for us it's sadly unusable.

Reply:
Why?

Reply:
b/c we can not determine why we cannot access it. Why not just return the token but without content when forIdentity is incorrect? Then we can figure out in the app if we have the right identity available to do the operation.

Reply:
The app will be able to use the adapted truncated reference once it's implemented, so we don't have to change things for the app if we can wait

Reply:
I have a bad feeling about relying completely on truncated references with the current impl. Maybe we should remove the UseCases that allow to load by id and key?

Reply:
> b/c we can not determine why we cannot access it. 

So if I understand you correctly, the use case is: 
"The user scans a QR code of a RealtionshipTemplate with `forIdentity=A` and tries to open it with identity `B`."
And you would like to show an error message like "The Relationship Template you are trying to open belongs to another identity". 
If that's the case, you could also show something like "The Relationship Template you are trying to open either does not exist or belongs to another identity". 

> Why not just return the token but without content when forIdentity is incorrect? Then we can figure out in the app if we have the right identity available to do the operation.

Because of the two reasons that are already mentioned above:
1. It would be inconsistent to how we do it for other entities like Messages or Relationships.
2. It would leak some information even though it's not 100% necessary, which might lead to security "problems".


Don't get me wrong, I'm not 100% against it. I just don't want to change it now and have to undo it a few days later because you realize there's a better way.

Reply:
> Since I don't know a use for loading by id and key, I'm in favour of it :)

@Magnus-Kuhn let's discuss this elsewhere and get back to Timo after that. I think this could be the solution, but I could be wrong.

Reply:
@tnotheis thanks for clarifying again :)